packages/runtime/container-runtime/src/summary/runningSummarizer.ts

```diff
@@ -780,25 +777,24 @@ export class RunningSummarizer extends TypedEventEmitter<ISummarizerEvents> impl
 			};
 			this.emit("summarize", eventProps);
 
-			// If the failure doesn't have "retryAfterSeconds" or the max number of attempts have been done, we're done.
-			if (retryAfterSeconds === undefined || currentAttempt >= maxAttempts - 1) {
-				done = true;
+			// If the failure doesn't have "retryAfterSeconds", we're done.
+			if (retryAfterSeconds === undefined) {
+				break;
 			}
 
 			// If the failure has "retryAfterSeconds", add a delay of that time. In this case, a final attempt will
 			// take place and we need to wait for "retryAfterSeconds" before that.
-			if (retryAfterSeconds !== undefined) {
-				this.mc.logger.sendPerformanceEvent({
-					eventName: "SummarizeAttemptDelay",
-					duration: retryAfterSeconds,
-					summaryNackDelay: ackNackResult.data?.retryAfterSeconds !== undefined,
-					stage: submitSummaryResult.data?.stage,
-					dynamicRetries: true, // To differentiate this telemetry from regular retry logic
-					...summarizeProps,
-				});
-				await delay(retryAfterSeconds * 1000);
-			}
-		} while (!done);
+			this.mc.logger.sendPerformanceEvent({
+				eventName: "SummarizeAttemptDelay",
+				duration: retryAfterSeconds,
+				summaryNackDelay: ackNackResult.data?.retryAfterSeconds !== undefined,
+				stage: submitSummaryResult.data?.stage,
+				dynamicRetries: true, // To differentiate this telemetry from regular retry logic
+				...summarizeProps,
+			});
+
+			await delay(retryAfterSeconds * 1000);
+		} while (currentAttempt < maxAttempts - 1); // We stop when we've got one last attempt to try
```

Comment:
I don't like that while it says we should keep running this block until this condition is satsified but its a lie. There are other conditions which can terminate the loop.
Anyway, I don't feel strongly about it. But you should update the comment to reflect it. For instance, this condition holds true only if the failures have `retryAfterSeconds` specified.